src/Fabulous.XamarinForms/Xamarin.Forms.Core.fs

```diff
@@ -1,446 +1,803 @@
-    let SearchBar = Widgets.register<Xamarin.Forms.SearchBar>()
-    let ToolbarItem = Widgets.register<Xamarin.Forms.ToolbarItem>()
+
+    let TapGestureRecognizer =
+        Widgets.register<Xamarin.Forms.TapGestureRecognizer>()
+
+    let SearchBar =
+        Widgets.register<Xamarin.Forms.SearchBar>()
+
+    let ToolbarItem =
+        Widgets.register<Xamarin.Forms.ToolbarItem>()
+
     let Editor = Widgets.register<Xamarin.Forms.Editor>()
-    let ViewCell = Widgets.register<Xamarin.Forms.ViewCell>()
-    let ImageButton = Widgets.register<Xamarin.Forms.ImageButton>()
-    let TabbedPage = Widgets.register<Xamarin.Forms.TabbedPage>()
-    let DatePicker = Widgets.register<Xamarin.Forms.DatePicker>()
+
+    let ViewCell =
+        Widgets.register<Xamarin.Forms.ViewCell>()
+
+    let ImageButton =
+        Widgets.register<Xamarin.Forms.ImageButton>()
+
+    let TabbedPage =
+        Widgets.register<Xamarin.Forms.TabbedPage>()
+
+    let DatePicker =
+        Widgets.register<Xamarin.Forms.DatePicker>()
+
     let TimePicker = Widgets.register<FabulousTimePicker>()
-    let Stepper = Widgets.register<Xamarin.Forms.Stepper>()
+
+    let Stepper =
+        Widgets.register<Xamarin.Forms.Stepper>()
 
 [<AbstractClass; Sealed>]
 type ViewBuilders private () =
     static member inline Application<'msg, 'marker when 'marker :> IPage>(mainPage: WidgetBuilder<'msg, 'marker>) =
-        ViewHelpers.buildWidgets<'msg, IApplication> ViewKeys.Application
-            [| Application.MainPage.WithValue(mainPage.Compile()) |]
-            
-    static member inline ContentPage<'msg, 'marker when 'marker :> IView>(title: string, content: WidgetBuilder<'msg, 'marker>) =
-        ViewHelpers.build<'msg, IContentPage> ViewKeys.ContentPage
-            [| Page.Title.WithValue(title) |]
-            [| ContentPage.Content.WithValue(content.Compile()) |]
-            [||]
-            
+        ViewHelpers.buildWidgets<'msg, IApplication>
+            ViewKeys.Application
+            [|
+                Application.MainPage.WithValue(mainPage.Compile())
+            |]
+
+    static member inline ContentPage<'msg, 'marker when 'marker :> IView>
+        (
+            title: string,
+            content: WidgetBuilder<'msg, 'marker>
+        ) =
+        WidgetBuilder<'msg, IContentPage>(
+            ViewKeys.ContentPage,
+            AttributesBundle(
+                StackList.one(Page.Title.WithValue(title)),
+                ValueSome [| ContentPage.Content.WithValue(content.Compile()) |],
+                ValueNone
+            )
+        )
+
     static member inline StackLayout<'msg>(orientation: Xamarin.Forms.StackOrientation, ?spacing: float) =
-        ViewHelpers.buildCollection<'msg, IStackLayout, IView> ViewKeys.StackLayout LayoutOfView.Children
-            [| StackLayout.Orientation.WithValue(orientation)
-               match spacing with None -> () | Some v -> StackLayout.Spacing.WithValue(v) |]
-            
+        match spacing with
```

Comment:
How can this one scale?
Like with multiple optional parameters?

Reply:
There is a ctor that accepts a collection.

Also as far as I can tell, optional args in inline functions are compiled away completely. So the code match `?spacing` can be messy and long. In release mode only the needed code will be left determined by calcite